cli/cmd/deploy/deploy.go

```diff
@@ -0,0 +1,397 @@
+package deploy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/fatih/color"
+	"github.com/go-git/go-git/v5"
+	"github.com/joho/godotenv"
+	"github.com/rilldata/rill/admin/client"
+	"github.com/rilldata/rill/cli/cmd/auth"
+	"github.com/rilldata/rill/cli/cmd/cmdutil"
+	"github.com/rilldata/rill/cli/cmd/org"
+	"github.com/rilldata/rill/cli/cmd/project"
+	"github.com/rilldata/rill/cli/pkg/config"
+	"github.com/rilldata/rill/cli/pkg/dotrill"
+	"github.com/rilldata/rill/cli/pkg/gitutil"
+	"github.com/rilldata/rill/cli/pkg/variable"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type options struct {
+	Name        string
+	Description string
+	ProdBranch  string
+	Public      bool
+	Variables   []string
+
+	// projectPath string input taken interactively
+	// no input taken from user for below
+	region   string
+	dbDriver string
+	dbDSN    string
+	slots    int64
+}
+
+// DeployCmd is the guided tour for deploying rill projects to rill cloud.
+// TODO :: add non interactive mode
+func DeployCmd(cfg *config.Config) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "deploy",
+		Short: "Guided tour for deploying rill projects to rill cloud",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			warn := color.New(color.Bold).Add(color.FgYellow)
+			info := color.New(color.Bold).Add(color.FgWhite)
+			success := color.New(color.Bold).Add(color.FgGreen)
+			var adminClient *client.Client
+
+			// log in if not logged in
+			if !cfg.IsAuthenticated() {
+				msg := fmt.Sprintf("In order to deploy to Rill Cloud, you must login. Opening your browsers to %s to login or sign up...", cfg.AdminURL)
+				warn.Println(msg)
+				// NOTE : calling commands within commands has both pros and cons
+				// PRO : No duplicated code
+				// CON : Need to make sure that UX under sub command verifies with UX on this command
+				loginCmd := auth.LoginCmd(cfg)
+				loginCmd.SetContext(ctx)
+				if err := loginCmd.RunE(loginCmd, nil); err != nil {
+					exitWithFailure(err)
+				}
+
+				// set token in config
+				token, err := dotrill.GetAccessToken()
+				if err != nil {
+					return fmt.Errorf("could not parse access token from ~/.rill: %w", err)
+				}
+				cfg.AdminTokenDefault = token
```

Comment:
Since `cfg` is passed by pointer, this can happen inside `LoginCmd` instead (I *think* Parag added that recently, but please check)